src/native/corehost/hostpolicy/deps_format.cpp

```diff
@@ -180,31 +180,153 @@ void deps_json_t::reconcile_libraries_with_targets(
     }
 }
 
-// Returns the RID determined (computed or fallback) for the platform the host is running on.
-pal::string_t deps_json_t::get_current_rid(const rid_fallback_graph_t& rid_fallback_graph)
+namespace
 {
-    pal::string_t currentRid = get_current_runtime_id(false /*use_fallback*/);
+    #define CURRENT_ARCH_SUFFIX _X("-") _STRINGIFY(CURRENT_ARCH_NAME)
+    #define RID_CURRENT_ARCH_LIST(os) \
+        _X(os) CURRENT_ARCH_SUFFIX, \
+        _X(os),
+
+    const pal::char_t* s_host_rids[] =
+    {
+#if defined(TARGET_WINDOWS)
+        RID_CURRENT_ARCH_LIST("win")
+#elif defined(TARGET_OSX)
+        RID_CURRENT_ARCH_LIST("osx")
+        RID_CURRENT_ARCH_LIST("unix")
+#elif defined(TARGET_LINUX_MUSL)
```

Comment:
For a non-portable build on musl Linux, we also want to add the target rid for probing.